doc/guide.md

```diff
@@ -2621,7 +2623,71 @@ the install operation will also deploy and un-deploy as
 necessary. This will only work if the ModuleDescriptor has the
 launchDescriptor property.
 
-### Purge
+### Install jobs and asynchronous operations
+
+For Okapi 4.2.0 and later, the install operation can be asynchronous.
+The asynchronous operation is enabled by URI parameter 'async=true'.
+As for the "synchronous" operation, the dependency check is performed
+first and install/upgrade will return 400 HTTP error upon failure.
+
+Following that, the install operation will create an install "job" on the
+server side and return HTTP status 201 along with a location of the newly
+created install job. The returned JSON content is defined by schema
+([InstallJob.json](../okapi-core/src/main/raml/InstallJob.json)).
+
+This location can then be inspected with HTTP GET for the progress of
+the install operation. The location is same base URI as install, but
+with a slash + the job ID. The install job has properties such as
+
+ * `complete`: boolean which tells whether the job has completed
+ * `id`: job id
+ * `date`: start time of job in UTC ISO8601 format.
+ * `modules`: enable/disable list of modules along with status.
+
+Each module entry is defined by schema
+([TenantModuleDescriptor.json](../okapi-core/src/main/raml/TenantModuleDescriptor.json)).
+Brief list of properties:
+
+ * `id`: module ID
+ * `from`: old module ID (absent if not upgrading)
+ * `action`: enable/disable/uptodate
+ * `stage`: the current stage of the module
+ * `message`: present upon error (error message)
+
+If `message` property is present an error has occurred and `stage`
+indicates at which stage the error occurred.
```

Comment:
Maybe mention and explain possible stages (deploy, call, and undeploy) here?

Reply:
ok